### PR TITLE
Only show first active care plan and goals/tasks for that plan

### PR DESCRIPTION
**Pull request by LUMacKenn**
### Ticket or Explanation for PR

[sc-2152]

I created separate state hooks (`activeGoals` and `activeTasks`) so that the goals and tasks can still be fetched async while fetching the careplan

### Please describe how the PR was tested

Web and mobile iOS

### PR merging checklist
Please add any other relevant checkbox items here.

- [x] I linked a Shortcut ticket to my PR (if applicable)
- [x] I pulled in the latest changes from `dev` and there's no conflicts
- [x] I corrected all linting/formatting issues
- [x] I tested my changes in a mobile emulator 
- [ ] My PR has been approved by at least one reviewer

### CODE REVIEW

File: capable/CarePlan.js [hunk 1]
```diff
@@ -80,7 +80,10 @@ class CarePlan extends DataWrapper {
             }),
         ]);
       } else throw "Care plan template not found.";
-    } else carePlan = new CarePlan(carePlans[0]);
+    } else {
+      const activeCarePlans = carePlans.filter((carePlan) => carePlan.status == "active");
+      carePlan = new CarePlan(activeCarePlans[0]);
```
Comment (jonfreeland) [inline capable/CarePlan.js:85]: Follow up with error handling for no active care plans.

File: capable/CarePlan.js [hunk 2]
```diff
@@ -80,7 +80,10 @@ class CarePlan extends DataWrapper {
             }),
         ]);
       } else throw "Care plan template not found.";
-    } else carePlan = new CarePlan(carePlans[0]);
+    } else {
+      const activeCarePlans = carePlans.filter((carePlan) => carePlan.status == "active");
```
Comment (jonfreeland) [inline capable/CarePlan.js:84]: Follow up with sorting by `created_at` descending to get the most recently created active care plan.

**Review by jonfreeland — APPROVED** [review]
Looks good. Noted a couple follow-ups.

**Review by jonfreeland — APPROVED** [review]

File: capable/CarePlan.js [hunk 3]
```diff
@@ -80,7 +80,17 @@ class CarePlan extends DataWrapper {
             }),
         ]);
       } else throw "Care plan template not found.";
-    } else carePlan = new CarePlan(carePlans[0]);
+    } else {
+      const activeCarePlans = carePlans
+        .filter((carePlan) => carePlan.status == "active")
```
Comment (VeraZab) [inline capable/CarePlan.js:85]: so on the barnard backend we have the ability to set careplans to active/inactive? Don't think I saw that..

**Review by VeraZab — APPROVED** [review]

File: capable/CarePlan.js (hunk 3 above)
Comment (LUMacKenn) [reply capable/CarePlan.js:85]: yup! dropdown in the portal

**Pull request merged**